photutils/geometry/triangular_overlap.pyx

```diff
@@ -0,0 +1,101 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+cimport numpy as np
+
+__all__ = ['triangular_overlap_grid']
+
+DTYPE = np.float64
+ctypedef np.float64_t DTYPE_t
+
+cimport cython
+
+def triangular_overlap_grid(double xmin, double xmax, double ymin,
+                            double ymax, int nx, int ny, double a,
+                            double b, double c, double theta,
+                            int use_exact, int subpixels):
+    """
+    triangular_overlap_grid(xmin, xmax, ymin, ymax, nx, ny,
+                            a, b, c, theta, use_exact, subpixels)
+
+    Area of overlap between a triangle and a pixel grid.
+
+    Parameters
+    ---------
+    xmin, xmax, ymin, ymax : float
+        Extent of the grid in the x and y direction
+    nx, ny : int
+        Grid dimensions
+    a, b, c : float
+        Sides of the triangle
+    theta : float
```

Comment:
I did start with allowing a free specification of the three corners, but I later realized that specifying an aperture just involves the shape and size of the aperture and the center of it is assumed to be the center of the extensions on the image, as in the case of rectangular aperture.

The three side lengths uniquely specify a triangle. Theta is used to specify the orientation of this triangle in space.

Does that make more sense?